cmake/std/atdm/common/toss3/environment_new.sh

```diff
@@ -0,0 +1,228 @@
+# export ATDM_CONFIG_CMAKE_JOB_POOL_LINK=2
+# NOTE: Above, currently setting CMAKE_JOB_POOL_LINK results in a build
+# failures with Ninja.  See https://gitlab.kitware.com/snl/project-1/issues/60
+
+# We do this twice since sems modules are wacked and we get errors to the screen on a purge
+# The second purge will catch any real errors with purging ...
+module purge &> /dev/null
+module purge
+. /projects/sems/modulefiles/utils/sems-modules-init.sh
+module load sems-env
+module load atdm-env
+module load atdm-ninja_fortran/1.7.2
+
+if [ "$ATDM_CONFIG_NODE_TYPE" == "OPENMP" ] ; then
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=8
+  export OMP_NUM_THREADS=2
+else
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=16
+fi
+
+if [ "$ATDM_CONFIG_COMPILER" == "INTEL-18.0.2_OPENMPI-2.0.3" ]; then
+    module load intel/18.0.2.199
+    module load openmpi-intel/2.0
+    module load mkl/18.0.5.274
+    module load sparc-cmake/3.12.3
+
+    export BOOST_ROOT=/projects/sparc/tpls/cts1-bdw/boost-1.65.1/00000000/cts1-bdw_intel-18.0.2
+    export CGNS_ROOT=/projects/sparc/tpls/cts1-bdw/cgns-c09a5cd/27e5681f1b74c679b5dcb337ac71036d16c47977/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export EUCLID_ROOT=/projects/sparc/tpls/cts1-bdw/euclid-19.30/95b8242ad729449fdb86591b9643c11463b61fa2/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export HDF5_ROOT=/projects/sparc/tpls/cts1-bdw/hdf5-1.10.5/00000000/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export LIBHIO_ROOT=/projects/sparc/tpls/cts1-bdw/libhio-1.4.1.2/00000000/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export METIS_ROOT=/projects/sparc/tpls/cts1-bdw/parmetis-4.0.3/00000000/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export NETCDF_ROOT=/projects/sparc/tpls/cts1-bdw/netcdf-4.7.0/58bc48d95be2cc9272a18488fea52e1be1f0b42a/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export PARMETIS_ROOT=/projects/sparc/tpls/cts1-bdw/parmetis-4.0.3/00000000/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export PNETCDF_ROOT=/projects/sparc/tpls/cts1-bdw/pnetcdf-1.10.0/6144dc67b2041e4093063a04e89fc1e33398bd09/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export SGM_ROOT=/projects/sparc/tpls/cts1-bdw/sgm-19.30/00000000/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+    export SUPERLUDIST_ROOT=/projects/sparc/tpls/cts1-bdw/superlu_dist-5.4.0/a3121eaff44f7bf7d44e625c3b3d2a9911e58876/cts1-bdw_intel-18.0.2_openmpi-2.0.3
+
+    export ATDM_CONFIG_SUPERLUDIST_INCLUDE_DIRS=${SUPERLUDIST_ROOT}/include
+    export ATDM_CONFIG_SUPERLUDIST_LIBS=${SUPERLUDIST_ROOT}/lib64/libsuperlu_dist.a
+    export ATDM_CONFIG_BINUTILS_LIBS="/usr/lib64/libbfd.so;/usr/lib64/libiberty.a"
+
+    export LD_LIBRARY_PATH=/usr/tce/packages/gcc/gcc-6.1.0/lib64:${LD_LIBRARY_PATH}
+elif [ "$ATDM_CONFIG_COMPILER" == "INTEL-19.0.5_OPENMPI-4.0.1" ]; then
+    module load intel/19.0.5.281
+    module load openmpi-intel/4.0
+    module load mkl/18.0.5.274
+    module load sparc-cmake/3.12.3
+
+    export CBLAS_ROOT=/projects/global/toss3/compilers/intel/intel_2019/compilers_and_libraries_2019.5.281/linux
```

Comment:
Is it not possible to reduce this duplication?  It seems like the only difference between between these paths  for Intel 18 and Intel 19 is the last part of the path containing either:

```
sparc_tpl_ext=cts1-bdw_intel-18.0.2
sparc_tpl_mpi_ext=cts1-bdw_intel-18.0.2_openmpi-2.0.3
```

for Intel 18 or:

```
sparc_tpl_ext=cts1-bdw_intel-19.0.5
sparc_tpl_mpi_ext=cts1-bdw_intel-19.0.5_openmpi-4.0.1
```

for Intel 19.

Then you could move all of the `export <TPLNAME>_ROOT=???` statements outside of the if statments and look like:

```
export BOOST_ROOT=/projects/sparc/tpls/cts1-bdw/boost-1.65.1/00000000/${sparc_tpl_ext}
...
export HDF5_ROOT=/projects/sparc/tpls/cts1-bdw/hdf5-1.10.5/00000000/${sparc_tpl_mpi_ext}
...
```

Reply:
When we update for the 'tlcc2' env, it will be easy to refactor to avoid duplication there as well.  (But we will wait into we get this PR done and merged before we bother with that.)